src/vs/workbench/contrib/terminal/test/browser/terminalProfileService.test.ts

```diff
@@ -0,0 +1,224 @@
+			'PowerShell': {
+				'source': ProfileSource.Pwsh,
+				'icon': 'terminal-powershell'
+			},
+			'Command Prompt': {
+				'path': [
+					'${env:windir}\\Sysnative\\cmd.exe',
+					'${env:windir}\\System32\\cmd.exe'
+				],
+				'args': [],
+				'icon': 'terminal-cmd'
+			},
+			'Git Bash': {
+				'source': ProfileSource.GitBash
+			},
+		},
+		linux: {
+			'bash': {
+				'path': 'bash',
+				'icon': 'terminal-bash'
+			},
+			'zsh': {
+				'path': 'zsh'
+			},
+			'fish': {
+				'path': 'fish'
+			},
+			'tmux': {
+				'path': 'tmux',
+				'icon': 'terminal-tmux'
+			},
+			'pwsh': {
+				'path': 'pwsh',
+				'icon': 'terminal-powershell'
+			}
+		},
+		osx: {
+			'bash': {
+				'path': 'bash',
+				'args': [
+					'-l'
+				],
+				'icon': 'terminal-bash'
+			},
+			'zsh': {
+				'path': 'zsh',
+				'args': [
+					'-l'
+				]
+			},
+			'fish': {
+				'path': 'fish',
+				'args': [
+					'-l'
+				]
+			},
+			'tmux': {
+				'path': 'tmux',
+				'icon': 'terminal-tmux'
+			},
+			'pwsh': {
+				'path': 'pwsh',
+				'icon': 'terminal-powershell'
+			}
+		}
+	}
+};
+
+suite('TerminalProfileService', () => {
+	let configurationService: TestConfigurationService;
+	let terminalProfileService: TestTerminalProfileService;
+	let remoteAgentService: TestRemoteAgentService;
+
+	setup(async () => {
+		configurationService = new TestConfigurationService({ terminal: { integrated: defaultTerminalConfig } });
+		remoteAgentService = new TestRemoteAgentService();
+
+		let instantiationService = new TestInstantiationService();
+		let terminalContributionService = new TestTerminalContributionService();
+		let extensionService = new TestExtensionService();
+		let localTerminalService = new TestLocalTerminalService();
+		let remoteTerminalService = new TestRemoteTerminalService();
+		let contextKeyService = new MockContextKeyService();
+
+		instantiationService.stub(IContextKeyService, contextKeyService);
+		instantiationService.stub(IExtensionService, extensionService);
+		instantiationService.stub(IConfigurationService, configurationService);
+		instantiationService.stub(IRemoteAgentService, remoteAgentService);
+		instantiationService.stub(ITerminalContributionService, terminalContributionService);
+		instantiationService.stub(ILocalTerminalService, localTerminalService);
+		instantiationService.stub(IRemoteTerminalService, remoteTerminalService);
+		instantiationService.stub(IWorkbenchEnvironmentService, TestEnvironmentService);
+		terminalProfileService = instantiationService.createInstance(TestTerminalProfileService);
+		localTerminalService.setProfiles([{
+			profileName: 'PowerShell',
+			path: 'C:Powershell.exe',
+			isDefault: true,
+			icon: ThemeIcon.asThemeIcon(Codicon.terminalPowershell)
+		}]);
+		remoteTerminalService.setProfiles([]);
+		terminalContributionService.setProfiles([{
+			extensionIdentifier: 'ms-vscode.js-debug-nightly',
+			icon: 'debug',
+			id: 'extension.js-debug.debugTerminal',
+			title: 'JavaScript Debug Terminal'
+		}]);
```

Comment:
Should all these be in the setup or in each individual test?

Reply:
deferring to the next PR in which I'll add more tests